frontend/public/graphql/client.ts

```diff
@@ -1,13 +1,47 @@
 import { ApolloClient } from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
 import { SubscriptionClient } from 'subscriptions-transport-ws';
 import { InMemoryCache } from 'apollo-cache-inmemory';
 import { WebSocketLink } from 'apollo-link-ws';
+import { split } from 'apollo-link';
 
 import { getK8sResourcePath } from '../module/k8s/resource';
 import { K8sKind, K8sResourceCommon } from '../module/k8s/types';
 import { URLQuery } from './client.gql';
 import { URLQueryType, URLQueryVariables } from '../../@types/gql/schema';
-import { getImpersonateHeaders } from '../co-fetch';
+import { getImpersonateHeaders, coFetch } from '../co-fetch';
+
+const isIOS = !!window.navigator.userAgent.match(/(iPhone|iPad)/);
+let wssErrors = 0;
+
+class GraphQLReady {
+  private callback: VoidFunction;
+  private ready: boolean;
+  private wasCalled: boolean;
+
+  setReady() {
+    this.ready = true;
+    if (!this.wasCalled && this.callback) {
+      this.wasCalled = true;
+      this.callback();
+    }
+  }
+
+  onReady(cb: VoidFunction) {
+    if (!isIOS) {
```

Comment:
Is there any performance penalty if we skip the iOS check and always fallback on WebSocket errors?

Reply:
There will be some difference but pretty small. And we will also gain the fallback for other cases.
I've removed the check, makes the code simpler and more secure which I think is worth the tiny perf downgrade.